### PR TITLE
get itemnames from vault

### PR DESCRIPTION
**Pull request by Mahagon**
New blocks like granite are shown as stone
vault should fix that. i made it softdepend so that its not a must have :)

### CODE REVIEW

**Comment by frymaster** [comment]
Interesting, I didn't know vault did that.

Currently the admin can override the names with their own choices. I'd like that behaviour preserved (ie try getting the names from within logblock **first**, and fall back to vault if it can't

**Comment by Mahagon** [comment]
custom names should always have the highes priority now :)

http://imgur.com/GAVDZ9w,fEsqx9T

**Comment by Mahagon** [comment]
just noticed: blockfilter is not compatible with that yet.
ill try to check that tomorrow

**Comment by frymaster** [comment]
committed in 8c232b59970cc959a2a3fe10cf58f2e6ece86927

**Comment by Mahagon** [comment]
uhm you just closed the wrong pull request :D 

-> #586

**Comment by frymaster** [comment]
Yes. Yes I did.

**Comment by Mahagon** [comment]
I just have no idea why but some lookups with blocks are messing around
searching for water adds ice 
searching for dirt adds farmland and grass...


Anyone has a idea? Everything else is running fine

for example /lb lookup block polished_diorite


here is some debug output:
    [23:12:09 INFO]: Mahagon issued server command: /lb lookup area 5 block 3:0 all
    [23:12:09 INFO]: types.add (with subid): 3:0 <- types.add just adds 1 Block
    [23:12:09 INFO]: ALL param: 3:0 <- in line 177 types contains 3 Blocks?!?!
    [23:12:09 INFO]: ALL param: 2:-1
    [23:12:09 INFO]: ALL param: 60:-1
    [23:12:20 INFO]: Mahagon issued server command: /lb lookup area 5 block 3 all
    [23:12:20 INFO]: types.add (without subid): 3:0
    [23:12:20 INFO]: ALL param: 3:0
    [23:12:20 INFO]: ALL param: 2:-1
    [23:12:20 INFO]: ALL param: 60:-1
    [23:14:04 INFO]: Mahagon issued server command: /lb lookup area 5 block podzol all
    [23:14:04 INFO]: types.add (without subid): 3:2
    [23:14:04 INFO]: ALL param: 3:2
    [23:14:04 INFO]: ALL param: 2:-1
    [23:14:04 INFO]: ALL param: 60:-1
    [23:19:07 INFO]: types.add (without subid): 8:0
    [23:19:07 INFO]: ALL param: 8:0
    [23:19:07 INFO]: ALL param: 9:-1
    [23:19:07 INFO]: ALL param: 79:-1
    [23:19:07 INFO]: getTable: [de.diddiz.util.Block@61bdbad9, de.diddiz.util.Block@61f259cb, de.diddiz.util.Block@3818c439]
    [23:19:16 INFO]: Mahagon issued server command: /lb lookup area 5 block 9 all
    [23:19:16 INFO]: types.add (without subid): 9:-1
    [23:19:16 INFO]: ALL param: 9:-1
    [23:19:16 INFO]: ALL param: 8:-1
    [23:19:16 INFO]: ALL param: 79:-1
    [23:19:16 INFO]: getTable: [de.diddiz.util.Block@2bb39aeb, de.diddiz.util.Block@6e15a63e, de.diddiz.util.Block@6c7b6c35]
    [23:19:22 INFO]: Mahagon issued server command: /lb lookup area 5 block 10 all
    [23:19:22 INFO]: types.add (without subid): 10:0
    [23:19:22 INFO]: ALL param: 10:0
    [23:19:22 INFO]: ALL param: 11:-1
    [23:19:22 INFO]: getTable: [de.diddiz.util.Block@3321ee3, de.diddiz.util.Block@74db9cfc]

**Comment by frymaster** [comment]
the MaterialName util class is only used for output, not querying

https://github.com/LogBlock/LogBlock/blob/master/src/main/java/de/diddiz/LogBlock/QueryParams.java#L581

That's the line that tries to look up a block type number from a string - it's using the Bukkit method.  This **will** need to be changed, now we're aware of it.

I'm just about to push a big update to the material names, so that class will be expanding in size shortly

**Comment by frymaster** [comment]
I pulled your data updates out of this commit and added them manually, then went through the wiki and added I think all of the missing ones.  The downside is of course it created conflicts with this PR

**Comment by frymaster** [comment]
Do you plan to resubmit this? It looks like a cool feature

On 25 February 2015 19:16:06 GMT+00:00, Daniel Jacobs <notifications@github.com> wrote:
>Closed #582.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/LogBlock/LogBlock/pull/582#event-241954659

-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Comment by Mahagon** [comment]
sure :D i just didnt want to spam travis 
but i found sth suspicious :o

a lookup for water or ice is returning ice, water and stationary water
same happens to dirt, farmland and grass
and lava + stationary lava

i searched a lot but i couldnt find the cause D:

i thought it was my fault but that its already in http://ci.md-5.net/job/LogBlock/227/

**Comment by frymaster** [comment]
I've pushed a method for the inverse - getting the type from a name - it'd be cool if your vault stuff integrated into that as well

File: pom.xml [hunk 1]
```diff
@@ -56,6 +55,11 @@
 			<artifactId>worldedit</artifactId>
 			<version>6.0.0-SNAPSHOT</version>
 		</dependency>
+		<dependency>
```
Comment (ammaraskar) [inline pom.xml]: If and when you update this, please add on a 
```xml
<scope>provided</scope>
```
here

**Comment by frymaster** [comment]
I was going to see about fixing the merge conflicts and fixing this up, but the branch related to this PR doesn't seem to exist - is this still something you want to do? It seems like it'd add value

**Comment by ammaraskar** [comment]
Does the vault repo even exist still?

**Comment by frymaster** [comment]
Looks like it changed at some point

http://nexus.theyeticave.net/content/repositories/pub_releases/net/milkbowl/vault/VaultAPI/

**Comment by Brokkonaut** [comment]
I will close this as outdated, and mojang/spigot fixed the item names so it should no longer be required

**Pull request closed without merging**